clippy_lints/src/hash_borrow_str_soundness.rs

```diff
@@ -0,0 +1,82 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::implements_trait;
+use rustc_hir::def::{DefKind, Res};
+use rustc_hir::{Item, ItemKind, Path, TraitRef};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::symbol::sym;
+
+declare_clippy_lint! {
+    ///
```

Comment:
```suggestion
    /// ### What it does
```
For consistency with all other lints

Reply:
I have cleaned up the explanation and added the header. I think the new one is better.

Reply:
Yeah, looks good to me now.